core/src/main/java/org/apache/iceberg/stats/BlobMetadata.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.stats;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import org.apache.iceberg.relocated.com.google.common.collect.ImmutableList;
+
+public class BlobMetadata {
+  private final String type;
+  private final List<Integer> columns;
+  private final long fileOffset;
+  private final long length;
+  private final Optional<String> compressionCodec;
+
+  @JsonCreator
```

Comment:
Because Iceberg is a library, we can't use annotations to control serialization. The JVM makes no guarantee that annotations are available at runtime and users are in control of the final classpath. So this brings in a new set of correctness problems. That's why we always serialize with a `SomethingParser` class.

Reply:
No worries! This is a problem most people aren't aware of so it's pretty common.